Observer/Config.php

```diff
@@ -7,24 +7,30 @@
 
 class Config implements ObserverInterface
 {
-
-    private $_helper;
-
     public function __construct(
-        \Metrilo\Analytics\Helper\Data $helper,
         \Magento\Framework\Message\ManagerInterface $messageManager,
-        \Metrilo\Analytics\Helper\AdminStoreResolver $resolver
+        \Metrilo\Analytics\Helper\Data              $dataHelper,
+        \Metrilo\Analytics\Helper\Activity          $activityHelper,
+        \Metrilo\Analytics\Helper\ApiClient         $apiClient
     ) {
         $this->messageManager = $messageManager;
-        $this->resolver = $resolver;
-        $this->_helper = $helper;
+        $this->dataHelper     = $dataHelper;
+        $this->activityHelper = $activityHelper;
+        $this->apiClient      = $apiClient;
     }
 
     public function execute(Observer $observer)
     {
-        $storeId = $this->resolver->getAdminStoreId();
-        if (!$this->_helper->createActivity($storeId, 'integrated')) {
-            $this->messageManager->addError('The API Token and/or API Secret you have entered are invalid. You can find the correct ones in Settings -> Installation in your Metrilo account.');
+        try {
+            $storeId  = $observer->getStore();
+            $activity = $this->activityHelper->createActivity($storeId, 'integrated');
+            $client   = $this->apiClient->getClient($storeId);
+    
+            if (!$client->createActivity($activity['url'], $activity['data'])) {
```

Comment:
this is weird, why do you create one of the activities via the `activityHelper` and the other one via the client directly?

Reply:
My naming is poor, maybe its better if i name the client method activityCall ($client->activityCall) since its a method responsible for the curl call to activityEndpoint.

Reply:
even if you do that, the question remains:
- Why are you doing the same thing (creating an activity) in two different ways?

Reply:
The first one ($this->activityHelper->createActivity) is generating the data (creating the data array for sync that is including the type, token and signature also combining the url. The second ($client->createActivity) is doing the api call with the generated data from the first. Reasons for generating the data in activity helper is for reusability, there should be two more activity calls during import witch are 'import_start' and 'import_end'.

Reply:
@adamof bump

Reply:
What is the point of having this in two separate calls? 
Why not just `$this->activityHelper->createActivity($storeId, 'integrated')` and do the sending in there?